tests/unit/SproxydKeysScan/SproxydKeysSubscribersTest.js

```diff
@@ -0,0 +1,35 @@
+const { setupProcessor } = require('../../utils/setupProcessor');
+const randomize = require('randomatic');
+const range = require('lodash/range');
+
+describe('SproxydKeysSubscribers', () => {
+    describe('DuplicateSproxydKeyFoundHandler', () => {
+        const [processor, duplicateHandler] = setupProcessor(10);
+
+        const key = 'test-2021-11-24T19-41-08-494Z/2\u000098362217123979999997RG001 ';
+        const version = () => `${randomize('0', 3)}.${randomize('0', 3)}`;
+
+        test('the lower (newer) version is always repaired and then set in the map', () => {
```

Comment:
Maybe only a test description issue: we need to repair the *higher (older)* version, not the lower (newer), because we don't want to repair the master version since it would require updating the master key as well and we ignore the master keys.

What we set in the map is indeed the lower (newer) version though, I think that is correct in the code and what should be done indeed to achieve the expected result (repairing all versions except the master/newer)